csharp/extractor/Semmle.Extraction/Extractor/BinaryLogExtractionContext.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Semmle.Util.Logging;
+
+namespace Semmle.Extraction
+{
+    public class BinaryLogExtractionContext : ExtractionContext
+    {
+        private readonly IEnumerable<SyntaxTree> generatedSyntaxTrees;
+        private readonly string compilationIdentifier;
+        private readonly string generatedFolderName;
+
+        public BinaryLogExtractionContext(string cwd, string[] args, string outputPath,
+            IEnumerable<SyntaxTree> generatedSyntaxTrees, string compilationIdentifier,
+            ILogger logger, PathTransformer pathTransformer, bool isQlTest)
+            : base(cwd, args, outputPath, [], logger, pathTransformer, ExtractorMode.BinaryLog, isQlTest)
+        {
+            this.generatedSyntaxTrees = generatedSyntaxTrees;
+            this.compilationIdentifier = compilationIdentifier;
+
+            // Compute a unique folder name for the generated files:
+            generatedFolderName = "generated";
+
+            if (Directory.Exists(generatedFolderName))
+            {
+                var counter = 0;
+                do
+                {
+                    generatedFolderName = $"generated{counter++}";
+                }
+                while (Directory.Exists(generatedFolderName));
+            }
+        }
+
+        private string? GetAdjustedPath(string path)
+        {
+            var syntaxTree = generatedSyntaxTrees.FirstOrDefault(t => t.FilePath == path);
+            if (syntaxTree is null)
+            {
+                return null;
+            }
+
+            return Path.Join(generatedFolderName, compilationIdentifier, path);
```

Comment:
This is most likely not enough to compute a unique path. There could be multiple project files with the same name in different folders in the source repo, and then the `compilationIdentifier`, which is in the following format: `test.csproj (net8.0)`, would not be unique.